config/routes.rb

```diff
@@ -78,8 +78,10 @@
     end
     resource :means_test_result, only: [:show]
     resource :declaration, only: %i[show update]
-    resources :dependants, only: %i[new create]
-    resources :dependant_details, only: %i[show update]
+
+    resources :dependants, only: %i[index create], controller: 'dependants/new' do
+      resource :details, only: %i[show update], controller: 'dependants/details'
+    end
```

Comment:
Comparing this with vehicles, I notice vehicles uses `resource :vehicle` rather than `resources :vehicles` and then a namespace, so perhaps the same pattern won't work. But try this:
```ruby
resources :dependants, only: %i[index create] do
  scope module: :dependants do
    resource :details, only: %i[show update]
  end
end
```
On my system that creates:
```bash
citizens_dependant_details GET   /citizens/dependants/:dependant_id/details(.:format) citizens/dependants/details#show
                           PATCH /citizens/dependants/:dependant_id/details(.:format) citizens/dependants/details#update
                           PUT   /citizens/dependants/:dependant_id/details(.:format) citizens/dependants/details#update
       citizens_dependants GET   /citizens/dependants(.:format)                       citizens/dependants#index
                           POST  /citizens/dependants(.:format)                       citizens/dependants#create
```